### PR TITLE
sys-apps/ipmicfg: New package

### PR DESCRIPTION
**Pull request by ConiKost**
sys-apps/ipmicfg: New package

Closes: https://bugs.gentoo.org/601318

### CODE REVIEW

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit versionator
```
Comment (mgorny) [inline sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: Please use `eapi7-ver` instead (it's forward-compat for EAPI 7 version functions).

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+RESTRICT="bindist fetch mirror strip"
+
+S="${WORKDIR}/${MY_PN}_${MY_PV}_build.${MY_DATE}"
+
+QA_PREBUILT="Linux/32bit/IPMICFG-Linux.x86 Linux/64bit/IPMICFG-Linux.x86_64"
+
+pkg_nofetch() {
+	elog "Please download ${A} from"
+	elog "${HOMEPAGE}/SwDownload/UserInfo.aspx?sw=0&cat=IPMI"
```
Comment (mgorny) [inline sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: Please don't ref HOMEPAGE like this. It is multi-value, and people may want to add extra sites to it.

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	elog "and place it in your DISTDIR directory."
+}
+
+src_install() {
+	# Choose ARCH
+	if use amd64; then
+		local MY_ARCH_BINARY="x86_64"
```
Comment (mgorny) [inline sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: Local variables go lowercase in bash.

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	# Install binary
+	exeinto "/opt/ipmicfg"
+	doexe "Linux/${MY_ARCH_FOLDER}/IPMICFG-Linux.${MY_ARCH_BINARY}"
+
+	# Install symlink
+	dodir "/opt/bin"
+	dosym "../ipmicfg/IPMICFG-Linux.${MY_ARCH_BINARY}" "/opt/bin/ipmicfg"
+
+	# Install docs
+	local DOCS="IPMICFG_UserGuide.pdf ReleaseNotes.txt"
```
Comment (mgorny) [inline sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: Array, please.

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild (hunk 2 above)
Comment (ConiKost) [reply sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: What to do instead? Write the full plain URL instead of a variable?

Comment (mgorny) [reply sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: Yes.

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild (hunk 1 above)
Comment (ConiKost) [reply sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: Changed to eapi7-ver by newest push.

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild (hunk 2 above)
Comment (ConiKost) [reply sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: URL is now in plain text by newest push.

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild (hunk 3 above)
Comment (ConiKost) [reply sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: Fixed by newest push.

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild (hunk 4 above)
Comment (ConiKost) [reply sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: Whoops, sorry. Fixed by newest push.

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+EAPI=6
+
+inherit eapi7-ver
+
+MY_DATE="$(ver_cut 4)"
+MY_PN="${PN^^}"
+MY_PV="$(ver_cut 1-3)"
+
+DESCRIPTION="IPMICFG is an in-band utility for configuring IPMI devices"
```
Comment (Zlogene) [inline sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: `DESCRIPTION` should not contain a package name

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+HOMEPAGE="https://www.supermicro.com"
+SRC_URI="ftp://ftp.supermicro.com/utility/${MY_PN}/${MY_PN}_${MY_PV}_build.${MY_DATE}.zip"
```
Comment (Zlogene) [inline sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: add `app-arch/unzip` to `DEPEND` as `SRC_URI` pointed to the `.zip` archive

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild (hunk 5 above)
Comment (ConiKost) [reply sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: Changed by newest push.

File: sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild (hunk 6 above)
Comment (ConiKost) [reply sys-apps/ipmicfg/ipmicfg-1.27.1.170901.ebuild]: Added by newest push.

**Comment by ConiKost** [comment]
@mgorny: I've changed the QA_PREBUILT. Do you know, how I do reproduce this QA warning? You told me yesterday about "-frecord-gcc-switches", which I have added to my CFLAGS, but still, I didn't got this warning.

**Comment by mgorny** [comment]
Looks good now, thanks!

**Pull request closed without merging**